ci/build.ps1

```diff
@@ -0,0 +1,125 @@
+Set-PSDebug -Trace 1
+
+$env:CONFIGURATION -match '^(?<compiler>\w+)_(?<bits>32|64)(?:-(?<option>\w+))?$'
+$compiler = $Matches.compiler
+$compileOption = $Matches.option
+$bits = $Matches.bits
+$cmakeBuildType = 'RelWithDebInfo'
+$depsCmakeVars = @{
+  CMAKE_BUILD_TYPE = $cmakeBuildType;
+}
+$nvimCmakeVars = @{
+  CMAKE_BUILD_TYPE = $cmakeBuildType;
+  BUSTED_OUTPUT_TYPE = 'nvim';
+  GPERF_PRG = 'C:\msys64\usr\bin\gperf.exe';
+}
+
+if ($compiler -eq 'MINGW')
+{
+  if ($bits -eq 32)
+  {
+    $arch = 'i686'
+  }
+  elseif ($bits -eq 64)
+  {
+    $arch = 'x86_64'
+  }
+  if ($compileOption -eq 'gcov')
+  {
+    $cmakeVars['USE_GCOV'] = 'ON'
+    $uploadToCodecov = $true
+  }
+  # These are native MinGW builds, but they use the toolchain inside
+  # MSYS2, this allows using all the dependencies and tools available
+  # in MSYS2, but we cannot build inside the MSYS2 shell.
+  $cmakeGenerator = 'MinGW Makefiles'
+  $cmakeGeneratorArgs = 'VERBOSE=1'
+
+  # Add MinGW to the PATH
+  $env:Path = "C:\msys64\mingw$bits\bin;$env:Path"
+  # Remove the Git sh.exe from the PATH
+  $env:Path = $env:Path.Replace('C:\Program Files\Git\usr\bin', '')
+
+  # Build third-party dependencies
+  C:\msys64\usr\bin\bash -lc "pacman --verbose --noconfirm -Su" ; exitIfFailed
+  C:\msys64\usr\bin\bash -lc "pacman --verbose --noconfirm --needed -S mingw-w64-$arch-cmake mingw-w64-$arch-perl mingw-w64-$arch-diffutils mingw-w64-$arch-unibilium gperf" ; exitIfFailed
+}
+elseif ($compiler -eq 'MSVC')
+{
+  $cmakeGeneratorArgs = '/verbosity:normal'
+  if ($bits -eq 32)
+  {
+    $cmakeGenerator = 'Visual Studio 15 2017'
+  }
+  elseif ($bits -eq 64)
+  {
+    $cmakeGenerator = 'Visual Studio 15 2017 Win64'
+  }
+}
+
+function convertToCmakeArgs($vars)
+{
+  return $vars.GetEnumerator() | foreach { "-D$($_.Key)=$($_.Value)" }
+}
+
+function exitIfFailed($command)
+{
+  if ($LastExitCode -ne 0)
+  {
+    exit $LastExitCode
```

Comment:
will this be clobbered by the `if ($LastExitCode -ne 0)` command?

Reply:
No, the [$LastExitCode](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_automatic_variables?view=powershell-6#lastexitcode) variable will only change after a Windows executable has exited.